src/collectors/github_rate_limit.rs

```diff
@@ -135,34 +140,47 @@ impl GitHubRateLimit {
         let u = client
             .execute(req)
             .await
-            .unwrap()
+            .map_err(Error::from)?
             .json::<GithubUser>()
             .await
-            .unwrap();
+            .map_err(Error::from)?;
 
-        u.login
+        Ok(u.login)
     }
 
     async fn update_stats(&mut self) {
+        debug!("Updating rate limit stats");
+
         #[derive(Debug, serde::Deserialize)]
         struct GithubRateLimit {
             pub rate: HashMap<String, usize>,
         }
 
         let client = reqwest::Client::new();
-
-        //FIXME: we will (might?) need a RWLock on users structure
         for u in self.users.iter_mut() {
-            let req = GithubReqBuilder::RateLimit
-                .build_request(&client, &u.token)
-                .unwrap();
-            let mut data = client
-                .execute(req)
-                .await
-                .unwrap()
-                .json::<GithubRateLimit>()
-                .await
-                .unwrap();
+            let req = match GithubReqBuilder::RateLimit.build_request(&client, &u.token) {
+                Ok(r) => r,
+                Err(e) => {
+                    error!("Unable to build request to update stats: {:?}", e);
+                    return;
+                }
+            };
+
+            let response = match client.execute(req).await {
+                Ok(resp) => resp,
+                Err(e) => {
+                    error!("Unable to execute request to update stats: {:?}", e);
+                    return;
+                }
+            };
+
+            let mut data = match response.json::<GithubRateLimit>().await {
+                Ok(d) => d,
+                Err(e) => {
+                    error!("Unable to deserialize rate limit stats: {:?}", e);
+                    return;
+                }
+            };
```

Comment:
Here I'd propagate the error with `?` (maybe adding some context), and then do a single match where this function is called.